Updated directions for installation in _windows.md (#3445)
dart-lang/site-www: src/tools/sdk/_windows.md

> How to install?
> 

Reviewer: I think we'd be fine without this, as the section of the page this is in is called "Installing the Dart SDK". 

```suggestion
```

You can install the Dart SDK using [Chocolatey.][Chocolatey]

> {{site.alert.important}} These commands require administrative privileges. 
> To start command prompt as an administrator following these steps:
> 1. Press <kbd>Windows+R</kbd> to open the **Run** window. 
> 2. Type “cmd” into the box. 
> 3. Press <kbd>Ctrl+Shift+Enter</kbd> to run the command as an administrator.
> 
> {{site.alert.end}}

Reviewer: I'm thinking since there are multiple ways to run commands as an administrator, through the old PowerShell and command prompt UIs as well as with the new Windows Terminal, we can present this as one example. I've also added some extra formatting. How does this look?

```suggestion
{{site.alert.important}} 
  These commands require administrative privileges. 

  For example, to start command prompt as an administrator
  following these steps:
  
  1. Press <kbd>Windows+R</kbd> to open the **Run** window. 
  
  2. Type `cmd` into the box. 
  
  3. Press <kbd>Ctrl+Shift+Enter</kbd> to run the command as an administrator.
  
{{site.alert.end}}
```

Author: looks great!

> ###How to install?

Reviewer: In this header, and others make sure to put a space after the final hash(`#`) or you'll end up with headers such as these:

![image](https://user-images.githubusercontent.com/18372958/128093098-f906d647-ba1c-4f32-a7e0-203da672cec4.png)

Also, I'm not sure we want to use headers within these OS specific instructions anyway, as then they will show up in the TOC even if a user does not have the Windows tab open. Then when clicking these with the Mac or Linux tab open, these TOC entries will not work(at least not properly). So perhaps we want to stick with the similar structure as before.

![image](https://user-images.githubusercontent.com/18372958/128093033-681ac558-aebd-4920-8c9d-e7b62496d510.png)

Author: Sure, will update.

You can install the Dart SDK using [Chocolatey.][Chocolatey]

> > **_NOTE:_** These commands require administrative privileges. To start command prompt as an administrator following these steps:

Reviewer: I think the previous `{{site.alert.important}}` structure matches the rest of the website better, but if you think it's highlighted too much you can use `{{site.alert.note}}` instead which introduces the Note text and icon.

Author: Ok i'll try that.

> 
> - Press Windows+R to open the “Run” box. 
> - Type “cmd” into the box 

> > - Press Ctrl+Shift+Enter to run the command as an administrator.

Reviewer: I agree that telling the user to search isn't the best solution, but I'm not sure this is as users may not want to use the old style run command prompt and may be using something like the new Windows Terminal.  I'll give some thought on how we can perhaps make this more generic.

Author: We can also add instructions for Windows Powershell which is the new Windows terminal. Although, only step 1 will change and rest will remain the same.

> > - Press Ctrl+Shift+Enter to run the command as an administrator.

Reviewer: For formatting keyboard keys and combinations, use `<kbd>` or backticks (`) as specified in https://developers.google.com/style/ui-elements?hl=en#keyboard

Author: Ok sure

####To install the Dart SDK:

```terminal
C:\> choco install dart-sdk
```

#####To upgrade the Dart SDK:

```terminal
C:\> choco upgrade dart-sdk
```

####Default installation directory:

```terminal
C:\tools\dart-sdk

> ```

Reviewer: Where did you find this? I don't have a Windows PC at the moment to try this, but I found on [Chocolatey's FAQ](https://docs.chocolatey.org/en-us/faqs#where-does-chocolatey-install-by-default) that it by default installs at:

> As of version 0.9.8.24, binaries, libraries and Chocolatey components install in C:\ProgramData\chocolatey (environment variable %ProgramData%) by default.

Author: Yes, that's the default path but not every package is installed there. In the same link, read under the heading: "Why doesn't a package install software to Program Files?". 

Apparently this is why everyone was having difficulty finding since everyone was trying to look inside program data/files.

I tested on my PC and it installed inside the tools folder.

###How to change installation directory?

> This package uses the help Get-BinRoot, which means that you can set an environment variable called ChocolateyToolsLocation, which controls where this package is extracted to. For more information click <a ahref=https://docs.chocolatey.org/en-us/create/functions/get-toolslocation>here</a>.

Reviewer: For this section and other parts of the document, we try to stick to under 80 characters long per line and use semantic line breaks to accomplish this. You can read more about how to do this here: https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks

Let me know if you have any questions about doing so.

Also for links, use the [Markdown link syntax](https://www.markdownguide.org/basic-syntax/#links) and try to create a more thoughtful statement instead of "click here". You can learn more about this in the documentation for [References to other documents](https://developers.google.com/style/cross-references?hl=en#references-to-other-documents).

Reviewer: Also I'm not super sure I understand the "help Get-BinRoot" piece of this instruction. What are you trying to convey there?

Author: Ok i'll do that.

And as for "Get-BinRoot", it's an inbuilt environment variable which Chocolatey uses for some packages which allows the user to change the installation directory. 

So our package dart-sdk uses that as I checked it on its page on Chocolatey. So that was just a minor detail about the package for people familiar with Chocolatey installations so that they find it easier to change the installation directory.

Reviewer: Thanks for the context. I'm not sure all the extra details here are necessary for a user to properly install Dart and may actually cause confusion. So is the essential idea that users can configure the `ChocolateyToolsLocation` environment variable to change the install location? If so, perhaps we can simply state something about that, and after that direct users with a statement like "For more information, see [Chocolatey's GET-TOOLSLOCATION documentation](https://docs.chocolatey.org/en-us/create/functions/get-toolslocation).".

Author: sure

###How to set path?

- In the Windows search bar type env, you will see "edit the system environment variables" option, open it
- Click on "Environment Variables...".
- Under user variables, select "Path" and click edit.
- Click new and add the path to your dart-sdk folder.

> - Press ok and apply.

Reviewer: A few notes:

- If this is sequential, perhaps you can use an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists). 
- For buttons and other UI elements, put their names in bold(**) instead of quotes: https://developers.google.com/style/ui-elements?hl=en#formatting
- Also match the capitalization of the UI elements

Author: Point 3 is already like that. I'll implement the first 2 pointers. Thanks!

C:\> choco install dart-sdk
```

To upgrade the Dart SDK:

```terminal
C:\> choco upgrade dart-sdk
```

Default installation directory:

```terminal
C:\tools\dart-sdk
```

How to change installation directory?

This package uses the help Get-BinRoot, which means that 
you can set an environment variable called ChocolateyToolsLocation, which 
controls where this package is extracted to. 
For more information check out [get-toolslocation].

> How to set path?

Reviewer: As a clarifying question before I review this section further, does choco not automatically add packages to the path?

Author: It does, but for some cases (as reported in the original issue) it was not being added. So I added this information as well just in case someone struggles with it. But yes, we can put this under a troubleshooting heading.

Reviewer: Thanks for including this then, it will definitely be helpful. 

Instead of starting this section with a question, how about we switch to some sort of troubleshooting statement like you mentioned. I'm not exactly sure how to structure or state this, do you have any ideas? It could be something basic like:

```suggestion
If you don't have access to the Dart executables,
add the extraction location to your PATH:
```

Author: Yes, looks cleaner now.

C:\> choco install dart-sdk
```

To upgrade the Dart SDK:

```terminal
C:\> choco upgrade dart-sdk
```

> Default installation directory:
> 
> ```terminal
> C:\tools\dart-sdk
> ```
> 
> How to change installation directory?
> 
> This package uses the help Get-BinRoot, which means that 
> you can set an environment variable called ChocolateyToolsLocation, which 
> controls where this package is extracted to. 
> For more information check out [get-toolslocation].

Reviewer: Okay, I gave this section some thought and perhaps we should use more of a prose structure to describe these pieces while also avoiding some of the technical details about Chocolatey that may confuse readers. How about something like this, assuming my understanding of the `ChocolateyToolsLocation` environment variable is correct.

```suggestion
By default, the SDK is installed at `C:\tools\dart-sdk`.

To change the default installation location:

Set the `ChocolateyToolsLocation` environment variable
to the desired extraction location.
To read more about how the installer uses this configuration, 
see [Chocolatey's Get-ToolsLocation reference][get-toolslocation].
```

Author: Yes that's correct! Definitely looks more clear now.
